src/modules/action/actionProduct.ts

```diff
@@ -0,0 +1,16 @@
+export function actionProduct(e: Event) {
+  const spanElement: HTMLElement = document.querySelector('.size-basket');
+  let sum = spanElement.textContent as unknown as number;
+  const target = e.target as HTMLElement;
+  if (target.classList.contains('marker')) {
+    sum = sum - 1;
+
+    sum === 0 ? (spanElement.textContent = '') : (spanElement.textContent = sum as unknown as string);
+    target.classList.remove('marker');
+  } else if (target.classList.contains('btn-basket')) {
+    // objectSave[0].sizeBasked = objectSave[0].sizeBasked + 1
+    sum = +sum + 1;
+    spanElement.textContent = sum as unknown as string;
+    target.classList.add('marker');
+  }
+}
```

Comment:
Делай поменьше `as string` особенно там, где это можно легко кастовать в строку. Также в тернарнике в первом if можно вынести общее, чтобы проще выглядело. 
```suggestion
export function actionProduct(e: Event) {
  const target = e.target as HTMLElement;
  const spanElement: HTMLSpanElement = document.querySelector('.size-basket');
  let sum = +spanElement.textContent;

  if (target.classList.contains('marker')) {
    sum--;
    spanElement.textContent = sum === 0 ? '' : String(sum);

    target.classList.remove('marker');
  } else if (target.classList.contains('btn-basket')) {
    sum++;
    spanElement.textContent = String(sum);
    target.classList.add('marker');
  }
}

```